### PR TITLE
Add support for suppressing project events

### PR DESCRIPTION
**Pull request by amcasey**
VS doesn't consume (e.g.) `syntaxDiag` events, so there's no reason to do the work to compute their payloads.

### CODE REVIEW

**Comment by amcasey** [comment]
VS appears to have begun to receive these events as of #19864.

File: src/server/server.ts [hunk 1]
```diff
@@ -535,6 +535,7 @@ namespace ts.server {
                 hrtime: process.hrtime,
                 logger,
                 canUseEvents,
+                suppressProjectEvents,
```
Comment (sheetalkamat) [inline src/server/server.ts]: Feels like this should be flags that indicates event to suppress given that this does not suppress ```ProjectInfoTelemetryEvent``` which is project event

Comment (amcasey) [reply src/server/server.ts]: In the interest of keeping the name intelligible, I decided not to incorporate that.  They also don't seem conceptually related to projects.

**Comment by amcasey** [comment]
Note that only versions of VS patched to pass this additional flag will benefit from the change.

File: src/server/server.ts (hunk 1 above)
Comment (sheetalkamat) [reply src/server/server.ts]: then may be rename it to onlyTelemetryEvent or something as such?

**Comment by sheetalkamat** [comment]
Please add test to verify this functionality

File: src/server/server.ts (hunk 1 above)
Comment (amcasey) [reply src/server/server.ts]: It's not only telemetry.  We also want TypingsInstaller events.

Comment (amcasey) [reply src/server/server.ts]: We could call it `suppressDiagnosticEvents`, but then I'd feel obliged to let `ProjectsUpdatedInBackgroundEvent` through (which is fine, it just requires a different implementation).

Comment (amcasey) [reply src/server/server.ts]: Changed to `suppressDiagnosticEvents`.

**Comment by amcasey** [comment]
@sheetalkamat Better?

**Comment by amcasey** [comment]
Apparently, it is unwise to use GH's merge tool.  I'll redo it locally.

**Pull request merged**